src/main/java/com/cgdp/library/service/BookService.java

```diff
@@ -0,0 +1,32 @@
+package com.cgdp.library.service;
+
+import com.cgdp.library.dto.book.BookDTO;
+import com.cgdp.library.dto.book.CreateBookRequestDTO;
+import com.cgdp.library.entity.AuthorEntity;
+import com.cgdp.library.entity.BookEntity;
+import com.cgdp.library.exceptions.NotFoundException;
+import com.cgdp.library.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class BookService {
+
+    private final BookRepository bookRepository;
+    private final ModelMapper modelMapper;
+    private final EntityManager entityManager;
+
+    public BookDTO createBook(CreateBookRequestDTO requestDTO) {
+        AuthorEntity authorEntity = entityManager.getReference(AuthorEntity.class, requestDTO.authorId());
```

Comment:
This `entityManager.getReference` doesn't work as I expected, it will still fetch the author and that's not so efficient, we should just create a method in the author service to check if the author exists, then here we should create an AuthorEntity manually and only fill the id field, it will be enough